### PR TITLE
 GitHub Actions: Update document-lint annotations

### PR DESCRIPTION
**Pull request by T0biii**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

fix annotations for document lint (currently the info "Ducment linting failed. Please fix the issues." is missing)

image

## PR Checklist

- [ ] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [ ] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [ ] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [ ] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Review by stephybun — COMMENTED** [review]
Hey @T0biii, looking at the GHA history this doesn't appear to be failing

image

File: .github/workflows/document-lint.yaml [hunk 1]
```diff
@@ -23,9 +23,11 @@ jobs:
         with:
           go-version-file: ./.go-version
       - run: bash scripts/gogetcookie.sh
-      - run: make document-lint
+      - name: run document lint
+        id: lint
+        run: make document-lint 
         continue-on-error: true
```
Comment (T0biii) [inline .github/workflows/document-lint.yaml:29]: because of this the ci is always green

**Comment by stephybun** [comment]
@T0biii I believe that is intentional: https://github.com/hashicorp/terraform-provider-azurerm/pull/23921

**Comment by T0biii** [comment]
i also think so this added back the annotian because they got skipt because the If Check dont Work anymore

**Comment by stephybun** [comment]
Thanks for looking into this @T0biii, but I'm going to go ahead and close this PR.

From memory and looking at the PR that introduced this GHA, this tool isn't supposed to fail the CI since it runs for *all* the resource documentation in the provider. This meant contributors would end up being notified of documentation issues that are unrelated to their PR, so I believe this is by design and shouldn't be changed.

**Comment by T0biii** [comment]
@stephybun maybe you misunderstood becuase this pr dont change the behavior of the CI. 
The CI will still success but the Annotation wasnt added because the if check no longer works so this pr fixes the if check
https://github.com/hashicorp/terraform-provider-azurerm/pull/27779/files#diff-4098ba327847b232484d8f0655caa98a7506131db90e47f54059a4931938e961R31 (Line 31)

bevor this pr:
image

with this pr:
image

**Pull request closed without merging**